library/pod-security-policy/seccompv2/template.yaml

```diff
@@ -0,0 +1,303 @@
+            messageExpression: |
+              variables.badContainerProfilesWithoutFiles.join("\n")
+          - expression: 'size(variables.badContainerProfilesWithFiles) == 0'
+            messageExpression: |
+              variables.badContainerProfilesWithFiles.join("\n")
+      - engine: Rego
+        source:
+          rego: |
+            package k8spspseccomp
+
+            import data.lib.exempt_container.is_exempt
+
+            violation[{"msg": msg}] {
+                not input_wildcard_allowed_profiles
+                allowed_profiles := get_allowed_profiles
+                container := input_containers[name]
+                not is_exempt(container)
+                result := get_profile(container)
+                not allowed_profile(result.profile, result.file, allowed_profiles)
+                msg := get_message(result.profile, result.file, name, result.location, allowed_profiles)
+            }
+
+            get_message(profile, _, name, location, allowed_profiles) = message {
+                profile != "Localhost"
+                message := sprintf("Seccomp profile '%v' is not allowed for container '%v'. Found at: %v. Allowed profiles: %v", [profile, name, location, allowed_profiles])
+            }
+
+            get_message(profile, file, name, location, allowed_profiles) = message {
+                profile == "Localhost"
+                message := sprintf("Seccomp profile '%v' with file '%v' is not allowed for container '%v'. Found at: %v. Allowed profiles: %v", [profile, file, name, location, allowed_profiles])
+            }
+
+            input_wildcard_allowed_profiles {
+                input.parameters.allowedProfiles[_] == "*"
+            }
+
+            input_wildcard_allowed_files {
+                input.parameters.allowedLocalhostFiles[_] == "*"
+            }
+
+            allowed_profile(_, _, _) {
+                input_wildcard_allowed_profiles
+            }
+
+            allowed_profile(profile, _, _) {
+                profile == "Localhost"
+                input_wildcard_allowed_files
+            }
+
+            # Simple allowed Profiles
+            allowed_profile(profile, _, allowed) {
+                profile != "Localhost"
+                profile == allowed[_]
+            }
+
+            # annotation localhost without wildcard
+            allowed_profile(profile, file, allowed) {
+                profile == "Localhost"
+                allowed[_] == sprintf("Localhost/%s", [file])
+            }
+
+            # The profiles explicitly in the list
+            get_allowed_profiles[allowed] {
+                allowed := input.parameters.allowedProfiles[_]
+            }
+
+            get_allowed_profiles[allowed] {
+                profile := input.parameters.allowedProfiles[_]
+                profile == "Localhost"
+                file := object.get(input.parameters, "allowedLocalhostFiles", [])[_]
+                allowed := sprintf("Localhost/%s", [file])
```

Comment:
rather than attempting string packing/unpacking, we can just mimic the structure of local host profiles, this will simplify comparison later on.

basically, `allowed := {"type": allowed, "localHostProfile": file}`

Similar for above.

Then `get_profile()` is also simplified, as is the comparison with localhost. We also are no longer dependent on our scratch string format being invalidated.